mmv1/products/datastream/api.yaml

```diff
@@ -369,3 +367,685 @@ objects:
+              - !ruby/object:Api::Type::NestedObject
+                name: 'singleTargetDataset'
+                exactly_one_of:
+                  - destination_config.0.bigquery_destination_config.0.single_target_dataset
+                  - destination_config.0.bigquery_destination_config.0.source_hierarchy_datasets
+                description: |
+                  A single target dataset to which all data will be streamed.
+                properties:
+                  - !ruby/object:Api::Type::String
+                    name: 'datasetId'
+                    required: true
+                    description: |
+                      Dataset ID in the format projects/{project}/datasets/{dataset_id}
+              - !ruby/object:Api::Type::NestedObject
+                name: 'sourceHierarchyDatasets'
+                exactly_one_of:
+                  - destination_config.0.bigquery_destination_config.0.single_target_dataset
+                  - destination_config.0.bigquery_destination_config.0.source_hierarchy_datasets
+                description: |
+                  Destination datasets are created so that hierarchy of the destination data objects matches the source hierarchy.
+                properties:
+                  - !ruby/object:Api::Type::NestedObject
+                    name: 'datasetTemplate'
+                    required: true
+                    description: |
+                      Dataset template used for dynamic dataset creation.
+                    properties:
+                      - !ruby/object:Api::Type::String
+                        name: 'location'
+                        required: true
+                        description: |
+                          The geographic location where the dataset should reside.
+                          See https://cloud.google.com/bigquery/docs/locations for supported locations.
+                      - !ruby/object:Api::Type::String
+                        name: 'datasetIdPrefix'
+                        description: |
+                          If supplied, every created dataset will have its name prefixed by the provided value.
+                          The prefix and name will be separated by an underscore. i.e. _.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'backfillAll'
+        exactly_one_of:
+          - backfill_all
+          - backfill_none
+        allow_empty_object: true
+        send_empty_value: true
+        description: |
+          Backfill strategy to automatically backfill the Stream's objects. Specific objects can be excluded.
+        properties:
+          - !ruby/object:Api::Type::NestedObject
+            name: 'mysqlExcludedObjects'
+            description: |
+              MySQL data source objects to avoid backfilling.
+            conflicts:
+              - backfill_all.0.postgresql_excluded_objects
```

Comment:
is that here since the field in proto is oneoff? https://source.corp.google.com/piper///depot/google3/google/cloud/datastream/v1/datastream_resources.proto;rcl=head;l=1077

Reply:
yes - `conflicts` means that you can specify _at most one_.
vs `exactly_one_of` which means that you _must_ specify one and only one.